packages/loopring_v3/contracts/impl/BasicDepositContract.sol

```diff
@@ -71,36 +68,33 @@ contract BasicDepositContract is IDepositContract, ReentrancyGuard
     function deposit(
         address from,
         address token,
-        uint    amount,
+        uint96  amount,
         bytes   calldata /*auxiliaryData*/
         )
         external
         override
         payable
         onlyExchange
-        returns (uint actualAmount, uint tokenIndex)
+        returns (uint96 actualAmount, uint tokenIndex)
     {
         // Check msg.value
         if (isETHInternal(token)) {
-            require(msg.value == amount, "INVALID_ETH_DEPOSIT");
+            require(msg.value == uint(amount), "INVALID_ETH_DEPOSIT");
         } else {
             require(msg.value == 0, "INVALID_TOKEN_DEPOSIT");
              // Transfer the tokens from the owner into this contract
             if (amount > 0) {
                 token.safeTransferFromAndVerify(
                     from,
                     address(this),
-                    amount
+                    uint(amount)
                 );
             }
         }
 
         uint exchangeBalanceBefore = exchangeBalance[token];
         // Keep track how many tokens are deposited in the exchange
-        uint exchangeBalanceAfter = exchangeBalanceBefore.add(amount);
-        // Make sure the total max amount per token in the exchange is capped
-        require(exchangeBalanceAfter <= MAX_TOTAL_TOKEN_BALANCE, "MAX_AMOUNT_REACHED");
```

Comment:
Would like to keep this check to keep circuits easier.

Reply:
Brecht, is it correct to say that each token's balance on layer-2 for each user should be <= MAX_TOTAL_TOKEN_BALANCE, and each deposit/withdrawal's total amount should also be <= MAX_TOTAL_TOKEN_BALANCE, but the total balance in a DepositContract can be > MAX_TOTAL_TOKEN_BALANCE?

Reply:
See also other comments on different places about this. But if the index can decrease I may have to implement more checks anyway so the onchain check may not be that useful any more...